src/state/hooks.ts

```
@@ -313,10 +314,16 @@ export function useSendAcross() {
         referrer && ethers.utils.isAddress(referrer)
           ? tagAddress(data, referrer)
           : data;
+
+      if (!await validateContractAndChain(depositBox.address, currentlySelectedFromChain.chainId, signer)) {
+        return {};
+      }
+      
```

This seems counterintuitive to me. Should we throw an error here? Why are we throwing an empty object?

---

That's what the surrounding code does when things aren't defined.

---

Yeah I'm sure you are, but the pattern seems odd to me. If this an error, empty object seems an odd way to say "something bad happened".

---

How do we normally deal with errors? Don't we just console.error them? If we throw, does that cause the screen to go white?

---

I mean ideally we catch them so we can handle them, even if ultimately right now we just log them. We do have an ErrorContext I'm pretty sure (or one can be added).

@daywiss wdyt?

---

Its a good point, but i think its out of scope of this fix and require a refactor across all the hooks. Imo id tackle this change in v2, but generally i think this sucks because you have to invent a way to propogate errors with hooks.

---

@Tulun are you cool with punting on this since this is the way it's already done in this app?